src/internal_events/statsd.rs

```diff
@@ -93,3 +96,29 @@ impl<T: std::fmt::Debug + std::fmt::Display> InternalEvent for StatsdSocketError
         );
     }
 }
+
+#[derive(Debug)]
+pub struct StatsdInvalidMetric<'a> {
+    pub value: &'a MetricValue,
+    pub kind: &'a MetricKind,
+}
+
+impl<'a> InternalEvent for StatsdInvalidMetric<'a> {
+    fn emit_logs(&self) {
+        warn!(
+            message = "Invalid metric sent; dropping event.",
```

Comment:
This message is somewhat ambigious:

1. "dropping event" is used in other events to indicate no further components in the pipeline will receive this event. In this case, since this is a sink, there is no way for other components to receive it anyway.

2. having said that, it is valid to say that the event got dropped _before_ sending it to the external Statsd service, but in that case, the "invalid metric sent" part is confusing, as I already noted in an earlier comment.

(the original `warn` log solved this by mentioning "invalid metric sent to statsd sink", which makes it explicit that we're talking about the event sent to the sink, not the eventual metric sent to the external statsd service)

cc @lukesteensen

Reply:
Yes, the original wording here was poor. I think a simple change from `sent` to `received` should clear it up.